pkg/reconciler/pipeline/pipeline.go

```diff
@@ -0,0 +1,289 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// If there's nothing to update, just return.
+	if reflect.DeepEqual(p.Status, desired.Status) {
+		return p, nil
+	}
+
+	// Don't modify the informers copy.
+	existing := p.DeepCopy()
+	existing.Status = desired.Status
+
+	return r.EventingClientSet.MessagingV1alpha1().Pipelines(desired.Namespace).UpdateStatus(existing)
+}
+
+func (r *Reconciler) reconcileChannel(ctx context.Context, channelName string, channelResourceInterface dynamic.ResourceInterface, p *v1alpha1.Pipeline) (*unstructured.Unstructured, error) {
+	c, err := channelResourceInterface.Get(channelName, metav1.GetOptions{})
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			newChannel, err := resources.NewChannel(channelName, p)
+			logging.FromContext(ctx).Error(fmt.Sprintf("Creating Channel Object: %+v", newChannel))
+			if err != nil {
+				logging.FromContext(ctx).Error(fmt.Sprintf("Failed to create Channel resource object: %s/%s", p.Namespace, channelName), zap.Error(err))
+				return nil, err
+			}
+			created, err := channelResourceInterface.Create(newChannel, metav1.CreateOptions{})
+			if err != nil {
+				logging.FromContext(ctx).Error(fmt.Sprintf("Failed to create Channel: %s/%s", p.Namespace, channelName), zap.Error(err))
+				return nil, err
+			}
+			logging.FromContext(ctx).Error(fmt.Sprintf("Created Channel: %s/%s", p.Namespace, channelName), zap.Any("NewChannel", zap.Any("NewChannel", newChannel)))
+			return created, nil
+		} else {
+			logging.FromContext(ctx).Error(fmt.Sprintf("Failed to get Channel: %s/%s", p.Namespace, channelName), zap.Error(err))
+			return nil, err
+		}
+	}
+	logging.FromContext(ctx).Error(fmt.Sprintf("Found Channel: %s/%s", p.Namespace, channelName), zap.Any("NewChannel", c))
+	return c, nil
+}
+
+func (r *Reconciler) reconcileSubscription(ctx context.Context, step int, p *v1alpha1.Pipeline) (*eventingv1alpha1.Subscription, error) {
+	expected := resources.NewSubscription(step, p)
+
+	subName := resources.PipelineSubscriptionName(p.Name, step)
+	sub, err := r.subscriptionLister.Subscriptions(p.Namespace).Get(subName)
+
+	// If the resource doesn't exist, we'll create it.
+	if apierrs.IsNotFound(err) {
+		sub = expected
+		logging.FromContext(ctx).Info(fmt.Sprintf("Creating subscription: %+v : SUBSCRIBERSPEC: %+v", sub, sub.Spec.Subscriber))
+		newSub, err := r.EventingClientSet.EventingV1alpha1().Subscriptions(sub.Namespace).Create(sub)
+		if err != nil {
+			// TODO: Send events here, or elsewhere?
+			//			r.Recorder.Eventf(p, corev1.EventTypeWarning, subscriptionCreateFailed, "Create Pipeline's subscription failed: %v", err)
+			return nil, err
+		}
+		return newSub, nil
+	} else if err != nil {
+		logging.FromContext(ctx).Error("Failed to get subscription", zap.Error(err))
+		// TODO: Send events here, or elsewhere?
+		//		r.Recorder.Eventf(p, corev1.EventTypeWarning, subscriptionCreateFailed, "Create Pipelines's subscription failed: %v", err)
```

Comment:
same here, the error should be added to the log

Reply:
I'm not sure I follow. What do you mean added to the log?